src/_nebari/upgrade.py

```diff
@@ -744,7 +749,27 @@ class Upgrade_2024_4_1(UpgradeStep):
     def _version_specific_upgrade(
         self, config, start_version, config_filename: Path, *args, **kwargs
     ):
-        rich.print("Ready to upgrade to Nebari version [green]2024.4.1[/green].")
+        # Default configuration for the node groups was added in this version. Therefore,
+        # users upgrading who don't have any specific node groups defined on their config
+        # file already, will be prompted and asked whether they want to include the default
+        if provider := config.get("provider", ""):
+            provider_full_name = provider_enum_name_map[provider]
+            if provider_full_name in config and "node_groups" not in config.get(
+                provider_full_name, {}
+            ):
+                try:
+                    default_node_groups = provider_enum_default_node_groups_map[
+                        provider
+                    ]
+                    continue_ = Prompt.ask(
+                        f"Would you like to include the default configuration for the node groups in [purple]{config_filename}[/purple]?",
+                        choices=["y", "N"],
+                        default="N",
+                    )
+                    if continue_ == "y":
+                        config[provider_full_name]["node_groups"] = default_node_groups
```

Comment:
Have you checked if this value is actually written to the nebari-config.yaml file?  If so, then LGTM!

Reply:
good point, they are matching: 
https://github.com/nebari-dev/nebari/blob/79b1218cb6cec95a9bbb96ad27889015a47d27d8/src/_nebari/stages/infrastructure/__init__.py#L536-L543

https://github.com/nebari-dev/nebari/blob/79b1218cb6cec95a9bbb96ad27889015a47d27d8/src/_nebari/stages/infrastructure/__init__.py#L558-L563

Reply:
@Adam-D-Lewis yes, the config is properly updated when accepting the changes.

@viniciusdc I'm not entirely sure what you mean here.